contracts/RRUtils.sol

```diff
@@ -152,4 +152,39 @@ library RRUtils {
 
         return false;
     }
+
+    function compareNames(bytes memory self, bytes memory other) internal  returns (int){
+        bool equal = self.equals(other);
+        if(equal){ return 0; }
+
+        uint off = 0;
+        uint otheroff = 0;
+        uint head;
+        uint otherhead;
+        uint len = labelCount(self, 0);
+        uint otherlen = labelCount(other, 0);
+
+        while (!equal) {
+            if(len >= otherlen){
+                head = headposition(self, off);
+                off = progress(self, off);
+            }else{
```

Comment:
If you make this a separate 'if', we can avoid a redundant trip around the loop when `len == otherlen`.